### PR TITLE
Fix Alabama itemized deductions

### LINKED ISSUE #3493: Fix Alabama itemized deductions
In the 0.584.3 version, much of the AL itemized deduction logic make no sense to me.   Here are the sources:

Screenshot 2023-12-18 at 3 28 11 PM

The first two items do make sense: `real_estate_taxes` are fully itemizable under the AL income tax and the `charitable_deduction` variable makes sense.

But after that, I'm puzzled.

For instance, what does the following comment mean?
```
# Redundant with `taxsim_tfica`, but broken out to keep in PolicyEngine variables.
```
What exactly is "broken out" and why?

And what is the `al_mortgage_interest_deduction`?  How exactly does it differ from the `mortgage_interest` variable?   And most importantly, why does the `al_mortgage_interest_deduction` variable have no formula.
And the same questions for the `al_investment_interest_deduction` variable.

It seems as if the Alabama itemized deduction calculation needs a comprehensive review and extensive revisions.

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #3493

copilot:all

### CODE REVIEW

File: policyengine_us/parameters/gov/states/al/tax/income/deductions/itemized/sources.yaml [hunk 1]
```diff
@@ -2,18 +2,21 @@ description: Alabama allows for the following itemized deductions.
...
     # Tax form allows deducting all contributions, not capped by AGI as in federal law.
     # We adopt the legal code definition.
     - charitable_deduction # Section (a), (10)
-    # Redundant with taxsim_tfica, but broken out to keep in PolicyEngine variables.
     - al_medical_expense_deduction # Section (a), (13)
+    - al_casualty_loss_deduction # Section (a), (6)
     - al_misc_deduction # Section (a), (23)
-    - al_mortgage_interest_deduction # Section (a), (2)
-    - al_investment_interest_deduction # Section (a), (2)
+    - mortgage_interest # Section (a), (2)
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/al/tax/income/deductions/itemized/sources.yaml:14]: @MaxGhenis 
Do we have a variable for just railroad retirement taxes (tier 1)
I know we use tax_unit_taxable_social_security but I am not sure if we can use that for exclusively railroad ret. benefits

image

File: policyengine_us/parameters/gov/states/al/tax/income/deductions/itemized/sources.yaml [hunk 2]
```diff
@@ -2,18 +2,21 @@ description: Alabama allows for the following itemized deductions.
...
+    - self_employment_tax # Section (a), (3), (a)
     # Legal code specifies charitable contributions to the extent deductible in federal law.
     # Tax form allows deducting all contributions, not capped by AGI as in federal law.
     # We adopt the legal code definition.
     - charitable_deduction # Section (a), (10)
-    # Redundant with taxsim_tfica, but broken out to keep in PolicyEngine variables.
     - al_medical_expense_deduction # Section (a), (13)
+    - al_casualty_loss_deduction # Section (a), (6)
     - al_misc_deduction # Section (a), (23)
-    - al_mortgage_interest_deduction # Section (a), (2)
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/al/tax/income/deductions/itemized/sources.yaml:12]: did you delete these variables too?

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/al/tax/income/deductions/itemized/sources.yaml:12]: now, yes

File: policyengine_us/variables/household/expense/person/investment_interest.py
Comment (MaxGhenis) [inline policyengine_us/variables/household/expense/person/investment_interest.py:1]: use existing interest_income

**Review by MaxGhenis — APPROVED** [review]
lgtm after this

**Pull request merged**